internal/errors/is.go

```diff
@@ -89,3 +89,20 @@ func IsMissingTableError(err error) bool {
 	}
 	return false
 }
+
+// IsNotFoundError returns a boolean indicating whether the error is known to
+// report a not found violation.
+func IsNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var domainErr *Err
+	if errors.As(err, &domainErr) {
+		if domainErr.Code == RecordNotFound {
+			return true
+		}
+	}
```

Comment:
Could this be simplified to:
```go
return errors.Is(err, ErrRecordNotFound)
```

Reply:
No, because not all `RecordNotFound` codes are the sentinel `ErrRecordNotFound`.  As I am doing this refactor I am also removing the dependence on the sentinels, at least I understood that to be one of the objectives...removing them does however cause this, @jimlambrt also previously brought up the benefit of wrapping the sentinel does allow for the `Is` check.  We have this comment in: https://github.com/hashicorp/boundary/blob/main/internal/errors/sentinels.go#L4
```
Creating new Sentinel type errors like these should be deprecated in favor of the new Err type that includes unique Codes and a Matching function.
```
If we are planning to not create new sentinels I see this as also wanting to remove dependence on old sentinels...

Reply:
Keep in mind also that As() will stop at the top level domain error while Is() will check every error down the wrapped list until it finds one that results in Is() being true or it reaches the end.